src/js/background.js

```diff
@@ -326,42 +328,34 @@ Badger.prototype = {
   },
 
   /**
-  * Checks a domain for the EFF DNT policy.
+  * Checks a domain for the EFF DNT policy if needed.
   *
   * @param {String} domain The domain to check
-  * @param {timestamp} nextUpdate Time when the DNT policy should be rechecked
   * @param {Function} cb Callback that receives check status boolean (optional)
   */
-  checkForDNTPolicy: function (domain, nextUpdate, cb) {
-    if (Date.now() < nextUpdate) {
-      // not yet time
-      return;
-    }
+  checkForDNTPolicy: function (domain, cb) {
+    let self = this;
 
-    var badger = this;
+    if (Date.now() < self.storage.getNextUpdateForDomain(domain)) {
+      return; // not yet time to re-check
+    }
+    if (self.checkedDNT.hasOrSet(domain)) {
+      return; // checked for DNT already and failed
+    }
 
-    if (! badger.isCheckingDNTPolicyEnabled()) {
-      // user has disabled this check
-      return ;
+    if (! self.isCheckingDNTPolicyEnabled()) {
+      return; // user has disabled this check
     }
 
     log('Checking', domain, 'for DNT policy.');
-
-    // update timestamp first;
-    // avoids queuing the same domain multiple times
-    var recheckTime = utils.getRandom(
-      utils.oneDayFromNow(),
-      utils.nDaysFromNow(7)
-    );
-    badger.storage.touchDNTRecheckTime(domain, recheckTime);
-
-    this._checkPrivacyBadgerPolicy(domain, function (success) {
+    self._checkPrivacyBadgerPolicy(domain, function (success) {
```

Comment:
Looking at it, maybe it's a good time to rename this called function to `_checkDNTPolicy` to make it more consistent.